kv/kvnemesis: TestKVNemesisMultiNode failed

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2413324&tab=buildLog) on [master@e25ee7ad4bcdeda49f40eb52c24d2d1b1aa67a86](https://github.com/cockroachdb/cockroach/commits/e25ee7ad4bcdeda49f40eb52c24d2d1b1aa67a86):

```
=== RUN   TestKVNemesisMultiNode
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode193332694
--- FAIL: TestKVNemesisMultiNode (2.78s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode193332694
    test_log_scope.go:63: use -show-logs to present logs inline
    kvnemesis_test.go:77: failure:
        error applying x.AdminChangeReplicas(ctx, /Table/50/"f00de566", [{ADD_REPLICA n4,s4}]) // snapshot failed: (n4,s4):4LEARNER: remote failed to apply snapshot for reason failed to apply snapshot: [n4,s4]: canApplySnapshotLocked: cannot add placeholder, have an existing placeholder range=37 [/Table/50/"ce435207"-/Max) (placeholder) (n2,s2):2: snapshot failed: (n4,s4):4LEARNER: remote failed to apply snapshot for reason failed to apply snapshot: [n4,s4]: canApplySnapshotLocked: cannot add placeholder, have an existing placeholder range=37 [/Table/50/"ce435207"-/Max) (placeholder) (n2,s2):2
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:627
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:362
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:72
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:99
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:73
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:909
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1357
        Wraps: (2) error applying x.AdminChangeReplicas(ctx, /Table/50/"f00de566", [{ADD_REPLICA n4,s4}]) // snapshot failed: (n4,s4):4LEARNER: remote failed to apply snapshot for reason failed to apply snapshot: [n4,s4]: canApplySnapshotLocked: cannot add placeholder, have an existing placeholder range=37 [/Table/50/"ce435207"-/Max) (placeholder) (n2,s2):2
        Wraps: (3) snapshot failed: (n4,s4):4LEARNER: remote failed to apply snapshot for reason failed to apply snapshot: [n4,s4]: canApplySnapshotLocked: cannot add placeholder, have an existing placeholder range=37 [/Table/50/"ce435207"-/Max) (placeholder) (n2,s2):2
          |
          | (opaque error leaf)
          | type name: github.com/cockroachdb/cockroach/pkg/roachpb/*roachpb.internalError
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueLeaf

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2419265&tab=buildLog) on [master@c456295f7d09fbd319509ae4b08fd63dae248ddf](https://github.com/cockroachdb/cockroach/commits/c456295f7d09fbd319509ae4b08fd63dae248ddf):

```
=== RUN   TestKVNemesisMultiNode
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode171211066
--- FAIL: TestKVNemesisMultiNode (3.14s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode171211066
    test_log_scope.go:63: use -show-logs to present logs inline
    kvnemesis_test.go:77: failure:
        error applying x.AdminChangeReplicas(ctx, /Table/50/"d7a26909", [{ADD_REPLICA n3,s3} {REMOVE_REPLICA n4,s4}]) // snapshot failed: (n3,s3):4LEARNER: remote failed to apply snapshot for reason failed to apply snapshot: [n3,s3]: canApplySnapshotLocked: cannot add placeholder, have an existing placeholder range=37 [/Table/50/"ba00a229"-/Max) (placeholder) (n2,s2):3: snapshot failed: (n3,s3):4LEARNER: remote failed to apply snapshot for reason failed to apply snapshot: [n3,s3]: canApplySnapshotLocked: cannot add placeholder, have an existing placeholder range=37 [/Table/50/"ba00a229"-/Max) (placeholder) (n2,s2):3
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:627
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:362
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:72
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:99
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:73
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:909
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1357
        Wraps: (2) error applying x.AdminChangeReplicas(ctx, /Table/50/"d7a26909", [{ADD_REPLICA n3,s3} {REMOVE_REPLICA n4,s4}]) // snapshot failed: (n3,s3):4LEARNER: remote failed to apply snapshot for reason failed to apply snapshot: [n3,s3]: canApplySnapshotLocked: cannot add placeholder, have an existing placeholder range=37 [/Table/50/"ba00a229"-/Max) (placeholder) (n2,s2):3
        Wraps: (3) snapshot failed: (n3,s3):4LEARNER: remote failed to apply snapshot for reason failed to apply snapshot: [n3,s3]: canApplySnapshotLocked: cannot add placeholder, have an existing placeholder range=37 [/Table/50/"ba00a229"-/Max) (placeholder) (n2,s2):3
          |
          | (opaque error leaf)
          | type name: github.com/cockroachdb/cockroach/pkg/roachpb/*roachpb.internalError
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueLeaf

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvnemesis/validator.go
+++ b/pkg/kv/kvnemesis/validator.go
@@ -354,6 +354,8 @@ func (v *validator) processOp(txnID *string, op Operation) {
 			// meantime, some other operation removed the same replica.
 		} else if resultIsError(t.Result, `remote failed to apply snapshot for reason failed to apply snapshot: raft group deleted`) {
 			// Probably should be transparently retried.
+		} else if resultIsError(t.Result, `remote failed to apply snapshot for reason failed to apply snapshot: .* cannot add placeholder, have an existing placeholder`) {
+			// Probably should be transparently retried.
 		} else if resultIsError(t.Result, `cannot apply snapshot: snapshot intersects existing range`) {
 			// Probably should be transparently retried.
 		} else if resultIsError(t.Result, `snapshot of type LEARNER was sent to .* which did not contain it as a replica`) {

